packages/hawtio/src/plugins/camel/camel-service.ts

```diff
@@ -397,3 +400,23 @@ export function isCamelVersionEQGT(node: MBeanNode, major: number, minor: number
 
   return compareVersions(camelVersion, major, minor) >= 0
 }
+
+const camelModelPromises = Promise.all([
+  import('@hawtio/camel-model-v3').then(c => {
+    _camel3 = c as unknown as CamelModel
+  }),
+  import('@hawtio/camel-model-v4').then(c => {
+    _camel4 = c as unknown as CamelModel
+  }),
+])
+
+function fetchCamelModules(): { camel3: CamelModel; camel4: CamelModel } {
+  if (!_camel3 || !_camel4) {
+    ;(async () => await camelModelPromises)()
+  }
+
+  return {
+    camel3: _camel3!,
+    camel4: _camel4!,
+  }
+}
```

Comment:
I think this async -> await doesn't work as you expect. Async call is blocked only inside if, which leads to null pointer exception as Paul mentioned because the if block finishes immediately and tries to return the values no matter what values `_camel3` and `_camel4` have.